[docs] Add component context to contributing guide

hickeyma opened the pull request:
Adds component context including references to tutorial.

This PR improves the usability around the process of creating a component for new users.

Partial https://github.com/open-telemetry/opentelemetry-collector/issues/5675

hickeyma committed:
Add component context to contributing guide

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
No changelog required.

fatsheep9146 commented on CONTRIBUTING.md:
```diff
@@ -75,10 +75,14 @@ providing the following information:
+* Provide a configuration structure which defines the configuration of the component
+* Provide the implementation which performs the component operation
+
+Familiarize yourself with the interface of the
```
There should not be a newline?

hickeyma replied:
Removed the newline

hickeyma committed:
Update after review

Review comments updated:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12490#discussion_r922929567
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12490#discussion_r923279371

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented on CONTRIBUTING.md:
```diff
@@ -75,10 +75,14 @@ providing the following information:
-Any component (receiver, processor, exporter, or extension) needs to implement the interfaces defined on the [core
-repository](https://github.com/open-telemetry/opentelemetry-collector). Familiarize yourself with the interface of the
-component that you want to write, and use existing implementations as reference.
+* Implement the `component.Component` interface
```
Added the link to Go doc

hickeyma commented:
@fatsheep9146 @mx-psi Thank you for the review. I have updated the comments and it is ready for review again.
